### PR TITLE
Add IErrorListHelper method to get rule id from handle

### PR DESCRIPTION
**Pull request by duncanp-sonar**
- moved the IErrorListHelper interface to Infrastructure.VS

Prep for #3594

### CODE REVIEW

File: src/Infrastructure.VS/IErrorListHelper.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        /// </remarks>
+        bool TryGetRuleIdFromSelectedRow(out SonarCompositeRuleId ruleId);
+
+        /// <summary>
+        /// Attempts to retrieve the Sonar rule key from the specific table entry in the Error List.
+        /// </summary>
+        /// The method will only return a rule key if the row represents a Sonar analysis issue for
+        /// any supported language (including Roslyn languages i.e. C# and VB.NET)
+
+        bool TryGetRuleId(ITableEntryHandle tableEntryHandle, out SonarCompositeRuleId ruleId);
```
Comment (duncanp-sonar) [inline src/Infrastructure.VS/IErrorListHelper.cs:45]: Users can click the "show help" hyperlink in the ErrorList "code" column without selecting the row first.

So, we need another method in this interface that doesn't check the selected rows, but instead accepts a handle indicating the row.

Unfortunately, that's a VS-specific interface, so I've moved `IErrorListHelper` to `Infrastructure.VS` :-(

File: src/Infrastructure.VS/IErrorListHelper.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.VisualStudio.Shell.TableControl;
+using SonarLint.VisualStudio.Core;
+
+namespace SonarLint.VisualStudio.Infrastructure.VS
```
Comment (duncanp-sonar) [inline src/Infrastructure.VS/IErrorListHelper.cs:24]: Changed the namespace

File: src/Infrastructure.VS/IErrorListHelper.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        /// <summary>
+        /// Attempts to retrieve the Sonar rule key from the selected item in the Error List.
+        /// </summary>
+        /// <remarks>
+        /// The method will only return a rule key if:
+        /// * there is a single row selected in the Error List
+        /// * the row represents a Sonar analysis issue for any supported language
+        ///   (including Roslyn languages i.e. C# and VB.NET)
+        /// </remarks>
+        bool TryGetRuleIdFromSelectedRow(out SonarCompositeRuleId ruleId);
```
Comment (duncanp-sonar) [inline src/Infrastructure.VS/IErrorListHelper.cs:37]: This method and comment are the same as before.

File: src/Infrastructure.VS.UnitTests/ErrorListHelperTests.cs [hunk 4]
```diff
@@ -74,7 +74,39 @@ public void GetErrorCode_SingleSonarIssue_ErrorCodeReturned(string fullRuleKey,
...
+        [DataRow("c:S111", "c", "S111", "SonarLint")]
+        [DataRow("cpp:S222", "cpp", "S222", "SonarLint")]
+        [DataRow("javascript:S333", "javascript", "S333", "SonarLint")]
+        [DataRow("typescript:S444", "typescript", "S444", "SonarLint")]
+        [DataRow("secrets:S555", "secrets", "S555", "SonarLint")]
+        [DataRow("foo:bar", "foo", "bar", "SonarLint")]
+        public void TryGetRuleId_FromHandle_ErrorCodeReturned(string fullRuleKey, string expectedRepo, string expectedRule, string buildTool)
+        {
+            // Note: this is a copy of TryGetRuleIdFromSelectedRow_SingleSonarIssue_ErrorCodeReturned,
+            //       but without the serviceProvider and IErrorList setup
```
Comment (duncanp-sonar) [inline src/Infrastructure.VS.UnitTests/ErrorListHelperTests.cs:89]: Exactly the same as the existing test, but with less setup.
I couldn't think of a good alternative to just copying the existing tests - let me know if you can.

Comment (bigfluffycookie) [reply src/Infrastructure.VS.UnitTests/ErrorListHelperTests.cs:89]: Hmm, I cant really think of  a better way either

**Review by bigfluffycookie — APPROVED** [review]

**Pull request merged**